Revert "Revert "Fix 388""

nderguti opened the pull request:
Reverts celements/celements-blog#20

https://synjira.atlassian.net/browse/CELDEV-388

nderguti committed:
Revert "Revert "Fix 388""

msladek reviewed (changes requested):
project has compile errors, please finish changes first before requesting reviews.

i think these errors came in due to a bad merge with the dev

nderguti committed:
Error fixed by NewsletterReciever

msladek committed:
format/cleanup

msladek committed:
Merge remote-tracking branch 'origin/dev' into revert-20-revert-16-fix_388

msladek commented on component/pom.xml:
```diff
@@ -36,7 +36,7 @@
       <groupId>com.celements</groupId>
       <artifactId>celements-core</artifactId>
-      <version>1.147</version>
+      <version>1.160</version>
```
update to current version 1.162

msladek commented on component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java:
```diff
@@ -414,89 +416,96 @@ private int parseGroupMembers(List<BaseObject> groupObjs, String type, XWikiCont
+            "xpage=celements_ajax&ajax_mode=BlogAjax&doaction=unsubscribe&emailadresse="
+                + URLEncoder.encode(emailAddresse, StandardCharsets.UTF_8.name()), getContext());
+      } catch (UnsupportedEncodingException e) {
+        LOGGER.error("UTF-8 unsupported.");
```
give exception a better name and add it to error logging

msladek commented on component/src/test/java/com/celements/blog/plugin/NewsletterReceiversTest.java:
```diff
@@ -19,18 +19,67 @@
+    comp = new NewsletterReceivers();
+  }
+
+  public void unsubscribeLink(String email) throws Exception {
```
private suffices

ebeutler commented on component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java:
```diff
@@ -414,89 +416,96 @@ private int parseGroupMembers(List<BaseObject> groupObjs, String type, XWikiCont
     } else {
       LOGGER.debug("No additional header. Doc does not exist: " + headerRef);
     }
+
```
please remove empty line

ebeutler commented on component/src/main/java/com/celements/blog/plugin/NewsletterReceivers.java:
```diff
@@ -414,89 +416,96 @@ private int parseGroupMembers(List<BaseObject> groupObjs, String type, XWikiCont
+
     LOGGER.debug("rendering content in " + renderLang);
-    context.setLanguage(renderLang);
+
```
please remove empty line

ebeutler reviewed (changes requested):
Just two empty lines to remove.

nderguti committed:
Updated to new version in pom.xml
Gave exception a better name. 
Private suffices for unsubscribeLink()
Removed empty lines
